### ISSUE TITLE
RSpec::Core::ExampleGroup::WrongScopeError: `expect` is not available on an example group

### ISSUE THREAD

**Issue by stadia** [rswag/rswag#63]
```
run_test! do |response|
    data = JSON.parse(response.body)
    expect(data['title']).to eq('foo')
end
```
scope is wrong and can't execute expect

**Comment by moserke** [comment]
Been having the same issue and could not get around it.  Best I could do is use `run_test!` for the default tests and then do the following for custom tests.  Just make sure `run_test!` is done first
```
  it 'custom test' do |example|
    expect().to
  end
```

**Comment by TLmaK0** [comment]
This is the way I do it:
```ruby
response '302', 'Solicitude without required fields with redirect_error' do
  let(:solicitude) { { solicitude: {}, redirect_error: 'http://testerror' } }
  header 'Location', type: :string, description: 'The redirection after post fails'

  before do |example|
    submit_request(example.metadata)
  end

  it 'redirects to error page' do |example|
    assert_response_matches_metadata(example.metadata) do |response|
      expect(response.header['Location']).to eq('http://testerror')
    end
  end
end
```

**Comment by lazebny** [comment]
We do it in this way (Rspec 3):
```ruby
response '302', 'Solicitude without required fields with redirect_error' do
  let(:solicitude) { { solicitude: {}, redirect_error: 'http://testerror' } }
  header 'Location', type: :string, description: 'The redirection after post fails'

  before do
    submit_request(example.metadata)
  end

  it 'redirects to error page' do
    assert_response_matches_metadata(example.metadata)
    expect(response.header['Location']).to eq('http://testerror')
  end
end
```

**Comment by TLmaK0** [comment]
@lazebny your way doesn't work for me with rspec 3.5 and rswag 1.3.0, gives me: `'example' is not available from within an example`

**Comment by domaindrivendev** [comment]
@stadia - the scoping issue should be resolved as of 1.5.0. It's also worth noting that the other workarounds provided are all perfectly valid approaches too.
